app/controllers/admin/service_announcements/toggle_controller.rb

```diff
@@ -0,0 +1,20 @@
+class Admin::ServiceAnnouncements::ToggleController < AdminController
+  before_action :set_service_announcement
+
+  # POST /admin/service_announcements/1/toggle
+  def create
+    respond_to do |format|
+      if @service_announcement.update active: !@service_announcement.active
```

Comment:
Is this actually a create endpoint? Why are you setting `active: !active`?

Reply:
I've been following http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/ so that each controller is only using the normal action methods, and they're broken up into little single responsibility controllers too.